tmt/steps/report/junit.py

```diff
@@ -111,36 +68,112 @@ class ReportJUnitData(tmt.steps.report.ReportStepData):
         help='Include full standard output in resulting xml file.')
 
 
+class ResultSummaryContext:
+    """
+    A class which keeps the results context (especially the result summary) for
+    JUnit template.
+    """
+
+    def __init__(self, results):
+        self._results = results
+
+    def __iter__(self):
+        return iter(self._results)
+
+    def __len__(self):
+        return len(self._results)
+
+    # TODO: Some better filtering function?
+    @functools.cached_property
+    def executed(self):
+        return [r for r in self._results if r.result != ResultOutcome.INFO]
+
+    @functools.cached_property
+    def skipped(self):
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self):
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self):
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self):
+        return sum([duration_to_seconds(r.duration) for r in self._results])
+
+
 @tmt.steps.provides_method('junit')
 class ReportJUnit(tmt.steps.report.ReportPlugin[ReportJUnitData]):
     """
-    Save test results in JUnit format.
+    Save test results in chosen JUnit flavor format. When flavor is set to
+    custom, the ``template_path`` with a path to a custom template must be
+    provided.
 
     When ``file`` is not specified, output is written into a file
     named ``junit.xml`` located in the current workdir.
     """
 
     _data_class = ReportJUnitData
 
+    def check_options(self) -> None:
+        """ Check the module options """
+
+        if self.data.flavor and self.data.template_path:
+            raise tmt.utils.ReportError("The options '--template-path' and "
+                                        "'--flavor' are mutually exclusive.")
+
+        self.data.flavor = self.data.flavor or DEFAULT_FLAVOR_NAME
```

Comment:
Please, add a `TODO` for me - we need a better way when looking for the source of option, I don't have the solution right now, but we already have some components. This `self.data.flavor or ...` is a task for `default` setting of the field itself, doing it in this imperative way is not going to be easy to maintain.

Reply:
I've taken a different approach by creating a flavor called `custom`. When this flavor is selected, the `--template-path` option must also be provided. Additionally, `--template-path` cannot be used with any other flavor.

I haven't found a better solution :/